src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/SerializerPass.php

```diff
@@ -47,11 +47,15 @@ private function findAndSortTaggedServices($tagName, ContainerBuilder $container
         }
 
         $sortedServices = array();
-        foreach ($services as $serviceId => $tags) {
-            foreach ($tags as $tag) {
-                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
-                $sortedServices[$priority][] = new Reference($serviceId);
+        foreach ($services as $serviceId => $attributes) {
+            $priority = 0;
+            foreach ($attributes as $tag) {
+                if (isset($tag['priority'])) {
+                    $priority = $tag['priority'];
+                    break;
+                }
             }
+            $sortedServices[$priority][] = new Reference($serviceId);
```

Comment:
Again, I think taking the `priority` from the index `0` tag is correct. But this fixes registering the service 2 times for 2 tags.